### DISCUSSION TITLE
High-availibility when the main server is down.

### DISCUSSION THREAD

**Discussion by KellyDi0te** [Ylianst/MeshCentral#5738] [Issues]
I have a 3 MeshCentral cluster with MongoDB databases and HAProxy with virtual IP.

The main question is : How do I make my agents go to server2 and server3 if the server1 is completely down and they don't know where to go anymore ? 

I did some tests with a virtual IP, everything is going fine until server1 goes down and now all the agents are going offline. 

I would like to use my virtual IP for the URI server in the agent configuration, but I'm not sure if this is the right way to do it :

![image](https://github.com/Ylianst/MeshCentral/assets/41836497/78dc9467-0be4-45cb-86ef-6d4eb618bbd5)

**Comment by si458** [comment]
You don't use IP addresses, you should use DNS names for starters...
And I think this is more of a HAProxy question than a meshcentral question.
You need to get HAProxy to check if the meshcentral port of server 1 is alive or dead, and if dead/noreply then it should mark it as down and move onto the next server
